core/blockchain.go

```diff
@@ -2103,7 +2140,7 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 		// memory here.
 		if len(blocks) >= 2048 || memory > 64*1024*1024 {
 			log.Info("Importing heavy sidechain segment", "blocks", len(blocks), "start", blocks[0].NumberU64(), "end", block.NumberU64())
-			if _, err := bc.insertChain(blocks, false); err != nil {
+			if _, err := bc.insertChain(blocks, false, nil); err != nil {
```

Comment:
Because in this part, the blocks are already inserted into the chain but the state is pruned so we need to reinsert. As the blocks are already inserted, sidecars are already inserted too.